packages/@aws-cdk/aws-apigateway/lib/authorizers/lambda.ts

```diff
@@ -0,0 +1,112 @@
+import iam = require('@aws-cdk/aws-iam');
+import lambda = require('@aws-cdk/aws-lambda');
+import { Construct, Duration, Lazy, Stack } from '@aws-cdk/core';
+import { CfnAuthorizer } from '../../lib/apigateway.generated';
+import { CustomAuthorizer } from '../../lib/authorizer';
+
+/**
+ * Properties for TokenAuthorizer
+ */
+export interface TokenAuthorizerProps {
+
+  /**
+   * An optional name for the authorizer. When provided, this will also be used for the physical id of the
+   * CloudFormation resource of type `AWS::ApiGateway::Authorizer`.
+   *
+   * @default - CDK will use the uniqueId assigned to this construct.
+   */
+  readonly name?: string;
+
+  /**
+   * The authorizer lambda function.
+   */
+  readonly function: lambda.IFunction;
+
+  /**
+   * The name of the header in the request that contains the authorization token as submitted by the client.
+   */
+  readonly headerName: string;
+
+  /**
+   * The TTL on how long APIGateway should cache the results. Max 1 hour.
+   * Disable caching by setting this to 0.
+   *
+   *  @default - 300 seconds
+   */
+  readonly resultsCacheTtl?: Duration;
+
+  /**
+   * An optional regex to be matched against the authorization token. When matched the authorizer lambda is invoked,
+   * otherwise a 401 Unauthorized is returned to the client.
+   *
+   * @default - no regex filter will be applied.
+   */
+  readonly validationRegex?: string;
+
+  /**
+   * An optional IAM role for APIGateway to assume before calling the Lambda-based authorizer.
+   *
+   * @default - A resource policy is added to the Lambda function allowing apigateway.amazonaws.com to invoke the function.
```

Comment:
Would it make sense that the Lambda resource policy will allow this role to invoke it if it is specified by the user?